index.bs

```diff
@@ -5679,39 +5674,34 @@ steps may throw an exception.
       <dt>If IsArray(|input|)</dt>
       <dd>
 
-        1. Let |len| be the ToLength(Get(|input|,
+        1. Let |len| be [=!=] ToLength(Get(|input|,
             "<code>length</code>"))
-        2. Assert: |len| will never an abrupt completion.
-        3. Add |input| to |seen|
-        4. Let |keys| be a new empty list
-        5. Let |index| be 0
-        6. While |index| is less than |len|, run these substeps:
-
-            1. Let |hop| be HasOwnProperty(|input|, |index|)
+        2. Add |input| to |seen|
+        3. Let |keys| be a new empty list
+        4. Let |index| be 0
+        5. While |index| is less than |len|, run these substeps:
 
-            2. Assert: |hop| is not an abrupt completion.
+            1. Let |hop| be [=!=] HasOwnProperty(|input|, |index|)
 
-            3. If |hop| is false, return invalid.
+            2. If |hop| is false, return invalid.
 
-            4. Let |entry| be |input|.[<span>[</span>Get]](|index|,
+            3. Let |entry| be [=?=] |input|.[<span>[</span>Get]](|index|,
                 |input|)
 
-            5. ReturnIfAbrupt(|entry|)
-
-            6. Let |key| be the result of running the steps to
+            4. Let |key| be the result of running the steps to
                 [=convert a value to a key=] with arguments |entry|
                 and |seen|
 
-            7. ReturnIfAbrupt(|key|)
+            5. ReturnIfAbrupt(|key|)
 
-            8. If |key| is invalid or an exception, then abort these
-                steps and return |key|.
+            6. If |key| is invalid abort these steps and return
+                invalid.
 
-            9. Append |key| to |keys|
+            7. Append |key| to |keys|
 
-            10. Increase |index| by 1
+            8. Increase |index| by 1
```

Comment:
Nit: these steps are inconsistent about whether they end with a period or not. (They should, generally.)